code/game/objects/structures/beds_chairs/pew.dm

```diff
@@ -0,0 +1,72 @@
+/obj/structure/chair/pew
+	name = "wooden pew"
+	desc = "Kneel here and pray."
+	icon = 'icons/obj/sofa.dmi'
+	icon_state = "pewmiddle"
+	resistance_flags = FLAMMABLE
+	max_integrity = 70
+	buildstacktype = /obj/item/stack/sheet/mineral/wood
+	buildstackamount = 3
+	item_chair = null
+
+/obj/structure/chair/pew/left
+	name = "left wooden pew end"
+	icon_state = "pewend_left"
+	var/mutable_appearance/leftpewarmrest
+
+/obj/structure/chair/pew/left/Initialize()
+	leftpewarmrest = GetLeftPewArmrest()
+	leftpewarmrest.layer = ABOVE_MOB_LAYER
+	return ..()
```

Comment:
you can just do `. = ..()`

`.` is the default return value of a proc

Reply:
gotcha, thanks for the info. I copied that syntax from comfy chairs.

Reply:
to be honest, I feel like there has to be a way to write that code in the pew section, and have it work in pew/left and pew/right, but I don't fully understand byond syntax all that well. I feel like having essentially identical code in pew/left and pew/right is dirty, but I wan't sure how to do it better.

Reply:
```DM
//under the base type
var/armrest_mut_appearance_key
var/mutable_appearance/the_armrest

//under the subtypes
armrest_mut_appearance_key = "pewend_left_armrest" //or "pewend_right_armrest" 

//Procs
/obj/structure/chair/pew/Initialize()
    . = ..()
    if(armrest_mut_appearance_key)
        the_armrest = mutable_appearance('icons/obj/sofa.dmi', armrest_mut_appearance_key)
        the_armrest.layer = ABOVE_MOB_LAYER

/obj/structure/chair/pew/Destroy()
    if(the_armrest)
        QDEL_NULL(the_armrest)
    return ..()

/obj/structure/chair/pew/post_buckle_mob(mob/living/M)
	. = ..()
	update_armrest()

/obj/structure/chair/pew/post_unbuckle_mob()
	. = ..()
	update_armrest()

/obj/structure/chair/pew/proc/update_armrest()
    if(the_armrest)
        if(has_buckled_mobs())
		    add_overlay(the_armrest)
	    else
		    cut_overlay(the_armrest)
```